### PR TITLE
Add/fix support for RID, Callable and Signal in editor properties

### LINKED ISSUE #62171: Editor crash when expanding array in debugger
### Godot version

v4.0.alpha10.official.4bbe7f0b9

### System information

Linux

### Issue description

When trying to inspect certain arrays (such as those containing a Callable) in the debugger panel, Godot crashes.

<details>
<summary> Backtrace </summary>

```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.alpha10.official (4bbe7f0b98de72d6dd77d5ade4b761de375bcf66)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x3e8e0) [0x7f391675e8e0] (??:0)
[2] /home/axel/Godot_v4.0-alpha10_linux.64() [0x1d7484f] (??:0)
[3] /home/axel/Godot_v4.0-alpha10_linux.64() [0x1d5a20f] (??:0)
[4] /home/axel/Godot_v4.0-alpha10_linux.64() [0x41ceb6a] (??:0)
[5] /home/axel/Godot_v4.0-alpha10_linux.64() [0x29986b9] (??:0)
[6] /home/axel/Godot_v4.0-alpha10_linux.64() [0x2998f3c] (??:0)
[7] /home/axel/Godot_v4.0-alpha10_linux.64() [0x2999327] (??:0)
[8] /home/axel/Godot_v4.0-alpha10_linux.64() [0x2916b35] (??:0)
[9] /home/axel/Godot_v4.0-alpha10_linux.64() [0x2937ad9] (??:0)
[10] /home/axel/Godot_v4.0-alpha10_linux.64() [0x295fb4c] (??:0)
[11] /home/axel/Godot_v4.0-alpha10_linux.64() [0x2960052] (??:0)
[12] /home/axel/Godot_v4.0-alpha10_linux.64() [0x296fbb2] (??:0)
[13] /home/axel/Godot_v4.0-alpha10_linux.64() [0xe3bd70] (??:0)
[14] /home/axel/Godot_v4.0-alpha10_linux.64() [0x3f8ad9f] (??:0)
[15] /home/axel/Godot_v4.0-alpha10_linux.64() [0x3f8bb95] (??:0)
[16] /home/axel/Godot_v4.0-alpha10_linux.64() [0xe3e6e9] (??:0)
[17] /home/axel/Godot_v4.0-alpha10_linux.64() [0xdc5b70] (??:0)
[18] /usr/lib/libc.so.6(+0x29290) [0x7f3916749290] (??:0)
[19] /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7f391674934a] (??:0)
[20] /home/axel/Godot_v4.0-alpha10_linux.64() [0xddfa3e] (??:0)
-- END OF BACKTRACE --
================================================================
```
</details>

### Steps to reproduce

1. Run the MRP in the editor. The project will pause due to a failed assert.
2. Down in the debugger panel, click the button that says "Array (size 3)"
image
3. The editor freezes for a moment and then crashes.

### Minimal reproduction project

[MRP_debugger_crash.zip](https://github.com/godotengine/godot/files/8933267/MRP_debugger_crash.zip)

```gdscript
extends Node2D

var foo = [1, 2, func(): print("lambda!")]

func _ready():
	assert(foo.size() > 3, "this assertion is supposed to fail")
```

### PR DESCRIPTION
**Pull request by MinusKube**
Callable and Signal editor properties were missing, causing the editor to crash when expanding an array containing one of them. Also, the RID editor property was always showing "Invalid RID" due to missing of serialization.

So, I added serialization for RID and Signal (but not Callable, since it's not always serializable when using custom callables), but maybe it was not the best thing to do?

Then, I added missing editor properties for Signal and Callable, as seen below:
Editor Properties

Fixes #62171

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]
Looks pretty good to me. Would be good if @reduz could review, and maybe @Faless for the marshalls changes. BTW, I would suggest reordering the `case`s there to match the same order as in `Variant::Type` (RID, Object, Callable, Signal).

**Comment by KoBeWi** [comment]
How does exported Callables behave? Are they not serialized, so upon load they are replaced with null?

It's possible to serialize them, but it's probably complicated to implement. Regular Callable is just object + method name. Then there is CallableCustomBind which is the same + array of arguments and CallableCustomUnbind, which is Callable + int. I think the only non-serializable callable that you can create in GDScript is lambda Callable.

File: core/io/marshalls.cpp [hunk 1]
```diff
@@ -610,11 +607,30 @@ Error decode_variant(Variant &r_variant, const uint8_t *p_buffer, int p_len, int
 			}
 
 		} break;
+		case Variant::RID: {
+			uint64_t id = decode_uint64(buf);
```
Comment (Faless) [inline core/io/marshalls.cpp:611]:
```suggestion
			ERR_FAIL_COND_V(len < 8, ERR_INVALID_DATA);
			uint64_t id = decode_uint64(buf);
```

File: core/io/marshalls.cpp [hunk 2]
```diff
@@ -610,11 +607,30 @@ Error decode_variant(Variant &r_variant, const uint8_t *p_buffer, int p_len, int
...
 		} break;
 		case Variant::SIGNAL: {
-			r_variant = Signal();
+			String name;
+			Error err = _decode_string(buf, len, r_len, name);
+			if (err) {
+				return err;
+			}
+
+			ObjectID id = ObjectID(decode_uint64(buf));
```
Comment (Faless) [inline core/io/marshalls.cpp:628]:
```suggestion
			ERR_FAIL_COND_V(len < 8, ERR_INVALID_DATA);
			ObjectID id = ObjectID(decode_uint64(buf));
```

**Review by Faless — CHANGES_REQUESTED** [review]

**Comment by MinusKube** [comment]
> How does exported Callables behave? Are they not serialized, so upon load they are replaced with null?
> 
> It's possible to serialize them, but it's probably complicated to implement. Regular Callable is just object + method name. Then there is CallableCustomBind which is the same + array of arguments and CallableCustomUnbind, which is Callable + int. I think the only non-serializable callable that you can create in GDScript is lambda Callable.

Yeah, it seems like since they're not serialized, they are replaced with empty Callables.

**Review by Faless — APPROVED** [review]
I think this is good to go now, the only problem I see is someone trying to transfer a signal across the network and trying to emit it, or sending a RID and expecting it to work on the other side (it won't of course).
But it seems safe otherwise, good work :+1: !

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**